### DISCUSSION TITLE
Authentication Module

### DISCUSSION THREAD

**Discussion by arantespp** [ttoss/modules#2] [Ideas]
### Description

Our first module will be the authentication module. We have a very opinionated process to perform authentication in our applications using [AWS Cognito](https://aws.amazon.com/cognito/). 

The authentication process needs the screens below:

![image](https://user-images.githubusercontent.com/16626980/118691099-1b631380-b7df-11eb-98dc-c979e4ef5e9b.png)

### API

The library will have some exports.

####  AuthProvider

Will handle the authentication state.

#### useAuth

This hook will provide some methods and states to handle our authentication flow.

**Comment by RayzaOliveira** [comment]
Hello @enniolopes . We recently created the `modules/auth` project, to unify the way we work with authentication, where both the interface part and the behavior is done in the `modules/auth` itself, thinking at this point we have some limitations regarding customizations, as the design is done in the component itself.

As we have our first project (siflor) that will use this module, and I saw that the design is a little different from what we developed in `modules/auth`, I would like to know if you believe we could maybe review the layout in modules or siflor, in regarding authentication. Since in the current design I only find the possibility to change the logo and colors of fonts and buttons.

PS: if you have any other ideas that we can adopt in this case, I'm available 😉

**Reply by enniolopes** [reply]
@RayzaOliveira you could change the Siflor UI to match our module.

**Reply by RayzaOliveira** [reply]
Perfect, as soon as I get back to the authentication part I'll make this adjustment.
